### PR TITLE
Support for other SRS authorities

### PR DESCRIPTION
**Pull request by gacarrillor**
and pass --defaultSrsAuth parameter to ili2db. 

Handles ili2db non-valid SRSs like USER:1234 (only valid for a single machine running QGIS) or srs codes that are not integers.

### CODE REVIEW

File: QgisModelBaker/gui/generate_project.py [hunk 1]
```diff
@@ -484,16 +490,24 @@ def update_crs_info(self):
...
-        if self.crsSelector.crs().authid()[:5] != 'EPSG:':
+        self.srs_auth = 'EPSG'  # Default
+        self.srs_code = 21781  # Default
+        srs_auth, srs_code = self.crsSelector.crs().authid().split(":")
+        if  srs_auth == 'USER':
             self.crs_label.setStyleSheet('color: orange')
             self.crs_label.setToolTip(
-                self.tr('Please select an EPSG Coordinate Reference System'))
-            self.epsg = 21781
+                self.tr('Please select a valid Coordinate Reference System.\nCRSs from USER are valid for a single computer and therefore, not recommended.\nA default EPSG:21781 will be used.'))
```
Comment (m-kuhn) [inline QgisModelBaker/gui/generate_project.py:499]: Are they "not recommended" or "will 21781 be used"?
From the description it looks like using USER crs'es should be supported but here it looks like they are not accepted. How does that work exactly?

File: QgisModelBaker/libili2db/ili2dbconfig.py [hunk 2]
```diff
@@ -225,8 +226,11 @@ def to_ili2db_args(self, extra_args=[], with_action=True):
         if self.create_basket_col:
             args += ["--createBasketCol"]
 
-        if self.epsg != 21781:
-            args += ["--defaultSrsCode", "{}".format(self.epsg)]
+        if self.srs_auth != 'EPSG':
+            args += ["--defaultSrsAuth", self.srs_auth]
+
+        if self.srs_code != 21781:
```
[the remark this answers is not in the document]
Comment (gacarrillor) [reply QgisModelBaker/libili2db/ili2dbconfig.py:232]: Yes! I think we should. I have seen those messages as well when testing our models with a default SRS. Not the case with Swiss models, though.

File: QgisModelBaker/gui/generate_project.py (hunk 1 above)
Comment (gacarrillor) [reply QgisModelBaker/gui/generate_project.py:499]: I'm 50%-50% here. The only issue I see is that when using a `USER:XXXX`, the SRS code is not guaranteed to remain in different QGIS installations. Other than that, I'm fine if we remove the `USER` conditional.

Comment (m-kuhn) [reply QgisModelBaker/gui/generate_project.py:499]: How about a warning (but letting the final choice to the user)? What does ili2db do with that anyway?

Comment (gacarrillor) [reply QgisModelBaker/gui/generate_project.py:499]: ili2db's schema import fails if the SRS is not found in the spatial reference table of the target DB.
I'm fine with the warning. I'll adjust it.

Comment (m-kuhn) [reply QgisModelBaker/gui/generate_project.py:499]: Let's remove the "not recommended" part. If ili2db does not support them we should not allow them.

Comment (gacarrillor) [reply QgisModelBaker/gui/generate_project.py]: Done!

File: QgisModelBaker/libili2db/ili2dbconfig.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (gacarrillor) [reply QgisModelBaker/libili2db/ili2dbconfig.py]: Fixed removing the if, ensuring the defaultSrsCode is always passed to ili2db in Schema Import.

**Pull request merged**